.github/workflows/regression-tests.yml

```diff
@@ -63,7 +61,34 @@ jobs:
           pytest --json-report --json-report-file=rep1_report.json -sv dragonfly/replication_test.py --df alsologtostderr --df enable_multi_shard_sync=true
           pytest --json-report --json-report-file=rep2_report.json -sv dragonfly/replication_test.py --df alsologtostderr --df enable_multi_shard_sync=false
 
+      - name: Run PyTests replication test with tls
+        if: ${{ matrix.runner  == 'ubuntu-latest' }}
+        timeout-minutes: 20
+        run: |
+          cd ${GITHUB_WORKSPACE}/tests
+          export DRAGONFLY_PATH="${GITHUB_WORKSPACE}/build/dragonfly" # used by PyTests
+
+          # We first need to generate the tls certificates to be used by the server
+
+          # Step 1
+          # Generate CA (certificate authority) key and self-signed certificate
+          # In production, CA should be generated by a third party authority
+          # Expires in one day and is not encrtypted (-nodes)
+          # X.509 format for the key
+          openssl req -x509 -newkey rsa:4096 -days 1 -nodes -keyout ca-key.pem -out ca-cert.pem -subj "/C=GR/ST=SKG/L=Thessaloniki/O=KK/OU=AcmeStudios/CN=Gr/emailAddress=acme@gmail.com"
+          # Self sign the certificate
+          openssl x509 -in ca-cert.pem -noout -text
+
+          # Step 2
+          # Generate Dragonfly's private key and certificate signing request (CSR)
+          openssl req -newkey rsa:4096 -nodes -keyout df-key.pem -out df-req.pem -subj "/C=GR/ST=SKG/L=Thessaloniki/O=KK/OU=Comp/CN=Gr/emailAddress=does_not_exist@gmail.com"
+
+          # Step 3
+          # Use CA's private key to sign dragonfly's CSR and get back the signed certificate
+          openssl x509 -req -in df-req.pem -days 1 -CA ca-cert.pem -CAkey ca-key.pem -CAcreateserial -out df-cert.pem -extfile df-ext.cnf
 
+          # Start the test
+          pytest --json-report --json-report-file=rep1_report.json -sv dragonfly/tls_replication_test.py --df alsologtostderr
```

Comment:
I should run these manually and verify.

Reply:
Can you hardcode the certificate into `tls_replication_test.py` and write the pem/crt into a temporary dir from there? I don't like this special pipeline and it makes running it locally harder.

Reply:
That's a really good idea. +1